src/OpaqueVal.h

```diff
@@ -86,12 +91,28 @@ class OpaqueMgr {
     std::unordered_map<std::string, Factory*> _types;
 };
 
-/** Macro to insert into an OpaqueVal-derived class's declaration. */
+/**
+ * Legacy macro to insert into an OpaqueVal-derived class's declaration. Overrides the "old" serialization methods
+ * DoSerialize and DoUnserialize.
+ * @deprecated Use DECLARE_OPAQUE_VALUE_V2 instead.
+ */
 #define DECLARE_OPAQUE_VALUE(T)                                                                                        \
     friend class zeek::OpaqueMgr::Register<T>;                                                                         \
     friend zeek::IntrusivePtr<T> zeek::make_intrusive<T>();                                                            \
-    std::optional<BrokerData> DoSerialize() const override;                                                            \
-    bool DoUnserialize(BrokerDataView data) override;                                                                  \
+    broker::expected<broker::data> DoSerialize() const override;                                                       \
+    bool DoUnserialize(const broker::data& data) override;                                                             \
+    const char* OpaqueName() const override { return #T; }                                                             \
+    static zeek::OpaqueValPtr OpaqueInstantiate() { return zeek::make_intrusive<T>(); }
+
+/**
+ * Macro to insert into an OpaqueVal-derived class's declaration. Overrides the "new" serialization methods
+ * DoSerializeData and DoUnserializeData.
+ */
+#define DECLARE_OPAQUE_VALUE_V2(T)                                                                                     \
```

Comment:
`_DATA` instead of `_V2`? Neither overly meaningful, but the former would be somewhat closer/related.